driver_android.go

```diff
@@ -28,15 +29,17 @@ static char* initAudioTrack(uintptr_t java_vm, uintptr_t jni_env,
   JavaVM* vm = (JavaVM*)java_vm;
   JNIEnv* env = (JNIEnv*)jni_env;
 
-  char hasAPI24 = 1;
+  bool hasAPI21 = true;
+  bool hasAPI24 = true;
   jclass android_media_AudioAttributes_Builder;
   jclass android_media_AudioFormat_Builder;
   jclass android_media_AudioAttributes;
 
   // check if the AudioAttributes.Builder subclass exists (since API 21, 5.0 Lollipop)
   jclass local = (*env)->FindClass(env, "android/media/AudioAttributes$Builder");
   if ((*env)->ExceptionCheck(env)) {
-    hasAPI24 = 0;
+    hasAPI21 = false;
+    hasAPI24 = false;
```

Comment:
This is theoretically correct, but I think resetting `hasAPI24` is not necessary here.

Reply:
There's a later statement `if (hasAPI24)` that would be incorrect if it is not set here. 

I'm thinking now, wouldn't it be better to just [query the build version](https://developer.android.com/reference/android/os/Build.VERSION#SDK_INT)?

Reply:
> There's a later statement if (hasAPI24) that would be incorrect if it is not set here.

I think the condition should also be fixed if we do. (probably `if (hasAPI21)`?)

> I'm thinking now, wouldn't it be better to just query the build version?

Ah yes, explicit querying the SDK version would be ideal.

Reply:
Ok I'll switch to explicitly querying the SDK. That makes more sense

Reply:
Updated